src/actions/Button.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useContext } from "react"
+import { NavigationContext } from "../global/NavigationContext"
+import Icon from "../icons/Icon"
+
+import { faArrowUpRightFromSquare } from "@fortawesome/free-solid-svg-icons"
+
+import "./Button.scss"
+
+export const isExternalLink = (href: string) => {
+  return href.startsWith("http://") || href.startsWith("https://")
+}
+
+export const isInternalLink = (href: string) => {
+  return href.startsWith("/") && !href.startsWith("//")
+}
+
+export interface ButtonProps {
+  /** Button content */
+  children: React.ReactNode
+  /** Appearance of the component */
+  variant?:
+    | "primary"
+    | "primary-outlined"
+    | "secondary"
+    | "secondary-outlined"
+    | "alert"
+    | "alert-outlined"
+    | "highlight" // or accent?
+    | "highlight-outlined" // or accent?
+    | "borderless-text"
+  /** Button size */
+  size?: "sm" | "md" | "lg"
+  /** Icon to show before the label text */
+  leadIcon?: React.ReactNode
+  /** Icon to show after the label text */
+  tailIcon?: React.ReactNode
+  /** Event handler for the button click */
+  onClick?: (e: React.MouseEvent) => void
+  /** Use an `<a href>` tag instead of `<button>` for a hyperlink */
+  href?: string
+  /** HTML button type */
+  type?: "button" | "submit" | "reset"
+  /** Set to true to disable the button */
+  disabled?: boolean
+  /** Set to true to hide the button from the accessibility tree */
+  ariaHidden?: boolean
+  /** Accessible label if button doesn't contain text content */
+  ariaLabel?: string
+  /** Element ID */
+  id?: string
+  /** Additional CSS classes */
+  className?: string
+}
+
+const setupButtonProps = (props: ButtonProps) => {
+  const classNames = ["seeds-button"]
+
+  const tailIcon =
+    props.href && isExternalLink(props.href) ? (
```

Comment:
I think this opens-in-a-new-tab needs to be optional based on current use cases

Reply:
Good point, I'll see if I come up with a good way to override or hide it. (Or are you saying it shouldn't show the icon by default?)

Reply:
@jaredcwhite I think we have more use cases without the icon, so I might recommend that by default it's not added, curious for @slowbot thoughts?

Reply:
I guess the question is, are the use cases without the icon supposed to change so they use the icon? Is including that icon the preferred best practice now?

Reply:
@slowbot Either - the previous issue was having no choice to turn it off, and then a second question of default behavior - on by default but able to be turned off sounds great!